### DISCUSSION TITLE
请问这个Graphscope与libgrape什么关系呀

### DISCUSSION THREAD

**Discussion by TerrenceYi** [alibaba/GraphScope#3139] [Q&A] [answered]
我之前看了下Graphscope的PR算法，对于度为0的顶点，将它的值赋值为base，其他顶点则是根据指向它的顶点赋值；而libgrape中的PR算法更符合我对PR算法的印象，请问这个Graphscope的PR算法的数学逻辑是什么呀，它能保证收敛并且收敛后的PR值之和为1吗

**Comment by sighingnow** [comment, marked as answer]
libgrape-lite is a submodule of the analytical engine in GraphScope (i.e., GAE).

The pagerank_networkx (https://github.com/alibaba/GraphScope/blob/main/analytical_engine/apps/pagerank/pagerank_networkx.h) in GraphScope is the networkx-compatible variant. And the `graphscope.pagerank` will invoke the one from libgrape-lite.

See also: https://github.com/alibaba/GraphScope/blob/main/python/graphscope/analytical/app/pagerank.py
